src/Entity/Reporting/EmailSubscriptionHistory.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace AppBundle\Entity\Reporting;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\ReferentTag;
+use Carbon\Carbon;
+use Doctrine\ORM\Mapping as ORM;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+/**
+ * @ORM\Table(
+ *      name="adherent_email_subscription_histories",
+ *      indexes={
+ *          @ORM\Index(name="adherent_email_subscription_histories_adherent_uuid_idx", columns="adherent_uuid"),
+ *          @ORM\Index(name="adherent_email_subscription_histories_adherent_action_idx", columns="action"),
+ *          @ORM\Index(name="adherent_email_subscription_histories_adherent_date_idx", columns="date"),
+ *          @ORM\Index(name="adherent_email_subscription_histories_adherent_email_type_idx", columns="subscribed_email_type"),
+ *      }
+ * )
+ * @ORM\Entity
+ *
+ * @Algolia\Index(autoIndex=false)
+ */
+class EmailSubscriptionHistory
+{
+    /**
+     * @var UuidInterface
+     *
+     * @ORM\Column(type="uuid")
+     * @ORM\Id
+     */
+    private $uuid;
+
+    /**
+     * Only UUID is used instead of the Entity because we cannot lose this data if one unsubscribes.
+     * Otherwise stats would be broken.
+     *
+     * @var UuidInterface
+     *
+     * @ORM\Column(type="uuid")
+     */
+    private $adherentUuid;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(length=50)
+     */
+    private $subscribedEmailType;
+
+    /**
+     * @var ReferentTag
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\ReferentTag")
+     */
+    private $referentTag;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(length=32)
+     */
+    private $action;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    private $date;
```

Comment:
This date should be immutable no ? he will not change in time ? Why using carbon here ?